code/web/src/modules/user/ProfileForm.js

```diff
@@ -0,0 +1,237 @@
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+// UI Imports
+import { Grid, GridCell } from '../../ui/grid'
+import Input from '../../ui/input/Input'
+import { H3, H4 } from '../../ui/typography'
+import Icon from '../../ui/icon'
+import Button from '../../ui/button'
+import { grey, grey2 } from '../../ui/common/colors'
+import { white } from "../../ui/common/colors"
+
+
+// App Imports
+import userRoutes from '../../setup/routes/user'
+import { updateUserInfo } from './api/actions'
+import { routeImage } from '../../setup/routes'
+import { renderIf } from '../../setup/helpers'
+import { upload, messageShow, messageHide } from '../common/api/actions'
+
+
+class ProfileForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userDetails: {
+        id: 0,
+        image: '',
+        email: this.props.user.details.email,
+        shippingAddress: this.props.user.details.address || '',
+        description: this.props.user.details.description || ''
+      },
+      isLoading: false
+    }
+  }
+
+  onChange = (event) => {
+    let userDetails = this.state.userDetails
+    userDetails[event.target.name] = event.target.value
+
+    this.setState({
+      userDetails
+    })
+  }
+
+  onUpload = (event) => {
+    this.props.messageShow('Uploading file, please wait...')
+
+    this.setState({
+      isLoading: true
+    })
```

Comment:
I know this is obvious, just want to make a note that we have to remember to set isloading to false once the post request returns.

Reply:
Good note Caleb!